deep_galaxy_training.py

```diff
@@ -102,17 +102,17 @@ def initialize(self):
 
                 # Bind a CUDA device to one MPI process (has no effect if GPUs are not used)
                 os.environ["CUDA_VISIBLE_DEVICES"] = str(hvd.local_rank())
-                
+
                 # Add callbacks
                 self.callbacks.append(hvd.callbacks.BroadcastGlobalVariablesCallback(0))
                 self.callbacks.append(hvd.callbacks.MetricAverageCallback())
                 self.callbacks.append(DataReshuffleCallback(self))
 
                 # Configure GPUs (if any)
-                gpus = tf.config.experimental.list_physical_devices('GPU')
-                for gpu in gpus:
-                    tf.config.experimental.set_memory_growth(gpu, self._gpu_memory_allow_growth)
-                    tf.config.experimental.set_visible_devices(gpu, 'GPU')
+                gpu = tf.config.experimental.list_physical_devices('GPU')[hvd.local_rank()]
```

Comment:
Thanks very much for submitting a PR, @puttsk ! I realize though that this line would crash if there is no GPU device, since in this case `tf.config.experimental.list_physical_devices` will return an empty array. Could you kindly fix that before I merge your PR? Thanks!

Reply:
I fixed this problem in commit https://github.com/maxwelltsai/DeepGalaxy/pull/1/commits/c8d0d26b0bedc790df6fb5bdb67d405b9e195deb.